Lib/pickle.py

```diff
@@ -203,19 +204,45 @@ def commit_frame(self, force=False):
             if f.tell() >= self._FRAME_SIZE_TARGET or force:
                 with f.getbuffer() as data:
                     n = len(data)
-                    write = self.file_write
-                    write(FRAME)
-                    write(pack("<Q", n))
-                    write(data)
+                    po = self.delayed_proto_opcode
+                    if po is None:
+                        self.file_write(FRAME + pack("<Q", n) + data)
+                    else:
+                        self.file_write(po + FRAME + pack("<Q", n) + data)
+                        self.delayed_proto_opcode = None
                 f.seek(0)
                 f.truncate()
 
+    def write_proto(self, data):
+        if self.current_frame:
+            # Store proto opcode to delay write when committing the first
```

Comment:
Can you explain why this is suddenly needed?

Reply:
I wanted to make the Python version issue the same number of calls to `file.write` as the C version does, that is only 1 call to write per 64kb frame. But this is probably a bad idea. Let me change that back.